src/utils/server-utils.ts

```diff
@@ -270,3 +278,119 @@ export async function getLayersAvailableDates(): Promise<AvailableDates> {
 
   return merge({}, ...layerDates);
 }
+
+/**
+ * Format value from featureInfo response based on LabelType provided
+ *
+ * @return a formatted string
+ */
+function formatFeatureInfo(value: string, type: LabelType): string {
+  if (type === LabelType.Date) {
+    return `${moment(value).utc().format('MMMM Do YYYY, h:mm:ss')} UTC`;
+  }
+
+  return value;
+}
+
+/**
+ * Executes a getFeatureInfo request
+ *
+ * @return object of key: string - value: string with formatted values given label type.
+ */
+async function runFeatureInfoRequest(
+  url: string,
+  wmsParams: requestFeatureInfo,
+  layers: WMSLayerProps[],
+): Promise<{ [name: string]: string }> {
+  // Transform to snake case.
+  const toSnakeWmsParams = Object.entries(wmsParams).reduce(
+    (obj, item) => ({
+      ...obj,
+      [item[0].replace(
+        /[A-Z]/g,
+        letter => `_${letter.toLowerCase()}`,
+      )]: item[1],
+    }),
+    {},
+  );
+
+  const res = await fetch(formatUrl(`${url}/ows`, toSnakeWmsParams));
+  const resJson: GeoJSON.FeatureCollection = await res.json();
+
+  const parsedProps = resJson.features.map(feature => {
+    // Get fields from layer configuration.
+    const [layerId] = (feature?.id as string).split('.');
+
+    const featureInfoProps =
+      layers?.find(l => l.serverLayerName === layerId)?.featureInfoProps || {};
+
+    const searchProps = Object.keys(featureInfoProps);
+
+    const properties = feature.properties ?? {};
+
+    return Object.keys(properties)
+      .filter(k => searchProps.includes(k))
+      .reduce(
+        (obj, key) => ({
+          ...obj,
+          [featureInfoProps[key].label]: formatFeatureInfo(
+            properties[key],
+            featureInfoProps[key].type,
+          ),
+        }),
+        {},
+      );
+  });
+
+  return parsedProps.reduce((obj, item) => ({ ...obj, ...item }), {});
+}
+
+/**
+ * This function builds and runs the getFeatureInfo request given the parameters
+ *
+ * @return Promise with returned object from request
+ */
+function fetchFeatureInfo(
+  layers: WMSLayerProps[],
+  url: string,
+  params: FeatureInfoType,
+): Promise<{ [name: string]: string }> {
```

Comment:
Don't define the return type and just let TS infer it. The plan (imo) is to strongly type the return of `runFeatureInfoRequest` and allow TS to infer the same type since you directly return the output of `runFeatureInfoRequest`